scripts/apps/relations/services/RelationsService.ts

```diff
@@ -2,6 +2,8 @@ import {pickBy, zipObject} from 'lodash';
 import {IArticle} from 'superdesk-api';
 import {isPublished} from 'apps/archive/utils';
 
+const RELATED_LINK_KEYS = 2;
```

Comment:
What does `2` and comparisons to it mean?

Reply:
it checks if association is a link - so it only has _id and type set (or only _id for BC)

Reply:
That's a sneaky one.

It's not a sustainable solution to identify it by the number of keys. Can you instead add a property `_internal: boolean` or similar and use that if you need to check it?

I don't mind if you do this in a separate PR. If you decide to keep `RELATED_LINK_KEYS` for now, add a comment describing the meaning behind the magic number :)

Reply:
what is BC by the way?

Reply:
backward compatibility

Reply:
Backward compatibility is technical debt :) One of the better ones out there, but it's still better to avoid it if possible. Which is possible in this case by writing an upgrade script to add `_internal` property. As I said in a previous comment, it's fine to do this in a separate PR to be able to release the fix faster.

Reply:
yep will fix that one separately